wsmaster/che-core-api-workspace/src/test/java/org/eclipse/che/api/workspace/server/spi/tck/WorkspaceDaoTest.java

```diff
@@ -104,7 +112,11 @@ public void createEntities() throws TckRepositoryException {
     workspaces = new WorkspaceImpl[COUNT_OF_WORKSPACES];
     for (int i = 0; i < COUNT_OF_WORKSPACES; i++) {
       // 2 workspaces share 1 namespace
-      workspaces[i] = createWorkspace("workspace-" + i, accounts[i / 2], "name-" + i);
+      if (i / COUNT_OF_WORKSPACES - 1 == 1) {
```

Comment:
I did not really get what you wanted to but it does not work:
```
    for (int i = 0; i < COUNT_OF_WORKSPACES; i++) {
     LoggerFactory.getLogger(WorkspaceDaoTest.class).info("Expression `i / COUNT_OF_WORKSPACES - 1 == 1`. i={}. Then `{} == 1` is {}", i, i / COUNT_OF_WORKSPACES - 1, i / COUNT_OF_WORKSPACES - 1 == 1);
     }
```
```
2019-04-10 17:41:22,334[main]             [INFO ] [e.c.a.w.s.s.t.WorkspaceDaoTest 116]  - Expression `i / COUNT_OF_WORKSPACES - 1 == 1`. i=0. Then `-1 == 1` is false
2019-04-10 17:41:22,362[main]             [INFO ] [e.c.a.w.s.s.t.WorkspaceDaoTest 116]  - Expression `i / COUNT_OF_WORKSPACES - 1 == 1`. i=1. Then `-1 == 1` is false
2019-04-10 17:41:22,363[main]             [INFO ] [e.c.a.w.s.s.t.WorkspaceDaoTest 116]  - Expression `i / COUNT_OF_WORKSPACES - 1 == 1`. i=2. Then `-1 == 1` is false
2019-04-10 17:41:22,365[main]             [INFO ] [e.c.a.w.s.s.t.WorkspaceDaoTest 116]  - Expression `i / COUNT_OF_WORKSPACES - 1 == 1`. i=3. Then `-1 == 1` is false
2019-04-10 17:41:22,366[main]             [INFO ] [e.c.a.w.s.s.t.WorkspaceDaoTest 116]  - Expression `i / COUNT_OF_WORKSPACES - 1 == 1`. i=4. Then `-1 == 1` is false
```

If you want to like create each second workspace from Devfile, then I guess it is not enough changes for WorkspaceDaoTest with such approach you just check a case with storing workspace with Devfile but not searching it, we definitely should have a separated tests for getting workspace by name with config and devfile
```java
  @Test
  public void shouldGetWorkspaceByNameInConfigAndNamespace() throws Exception {
    final WorkspaceImpl workspace = workspaces[0];

    assertEquals(
        workspaceDao.get(workspace.getConfig().getName(), workspace.getNamespace()),
        new WorkspaceImpl(workspace));
  }

  @Test
  public void shouldGetWorkspaceByNameInWorkspaceAndNamespace() throws Exception {
    final WorkspaceImpl workspace = workspaces[1];

    assertEquals(
        workspaceDao.get(workspace.getConfig().getName(), workspace.getNamespace()),
        new WorkspaceImpl(workspace));
  }
```
One more test case that should be tested:
```java

  @Test(expectedExceptions = ConflictException.class, 
  expectedExceptionsMessageRegExp = "Such name exist")
  public void shouldThrowExceptionIfNameIsNotUnique() throws Exception {
    final WorkspaceImpl workspace1 = createWorkspaceFromConfig("workspace-name", ...);
    final WorkspaceImpl workspace2 = createWorkspaceFromDevfile("workspace-name",...);

    workspaceDao.create(workspace1);
    workspaceDao.create(workspace2);
  }
```
^ Please do not just copy/paste because I've written it as example and it may be rewrtitten in nicer way, I guess.